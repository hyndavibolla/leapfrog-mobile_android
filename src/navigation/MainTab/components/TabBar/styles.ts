import { css } from '@emotion/native';

import { COLOR } from '../../../../constants';

export const styles = {
  container: css`
    flex-direction: row;
    align-items: center;
    justify-content: space-around;
    background-color: ${COLOR.WHITE};
    padding: 0 8px 8px;
  `
};
