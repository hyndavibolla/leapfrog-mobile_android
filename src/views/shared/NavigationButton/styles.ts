import { css } from '@emotion/native';
import { COLOR, FONT_SIZE } from '../../../constants';

export const styles = {
  container: css`
    align-self: flex-start;
  `,
  buttonContainer: css`
    width: 75px;
    height: 75px;
    align-items: center;
    justify-content: center;
  `,
  active: css`
    fill: ${COLOR.PRIMARY_BLUE};
  `,
  title: css`
    opacity: 0.5;
    margin-top: 8px;
    font-size: ${FONT_SIZE.TINY};
    color: ${COLOR.DARK_GRAY};
  `,
  textActive: css`
    color: ${COLOR.PRIMARY_BLUE};
  `
};
