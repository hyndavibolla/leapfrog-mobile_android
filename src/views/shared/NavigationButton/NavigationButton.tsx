import React, { memo, useMemo } from 'react';
import { TouchableHighlight, GestureResponderEvent, TouchableHighlightProps, View } from 'react-native';
import { Text } from '../Text';
import { View as AnimatedView } from 'react-native-animatable';

import { useTestingHelper } from '_utils/useTestingHelper';
import { Icon } from '_commons/components/atoms/Icon';
import { ICON } from '_constants/icons';
import { COLOR, FONT_SIZE } from '_constants/styles';

import { styles } from './styles';

export enum NavigationButtonType {
  STREAK = 'streak',
  EARN = 'earn',
  REWARD = 'reward',
  WALLET = 'wallet'
}

export interface Props extends TouchableHighlightProps {
  type: NavigationButtonType;
  active: boolean;
  onPress: (event: GestureResponderEvent) => void;
}
export const NavigationButton = ({ type, active, onPress, ...props }: Props) => {
  const { getTestIdProps } = useTestingHelper('navigation-button');
  const contentMap = useMemo(
    () => ({
      [NavigationButtonType.STREAK]: {
        title: 'Missions',
        icon: <Icon name={ICON.MISSION} size={FONT_SIZE.HUGE} />,
        inactiveIcon: <Icon name={ICON.MISSION} size={FONT_SIZE.HUGE} color={COLOR.DARK_GRAY} />
      },
      [NavigationButtonType.EARN]: {
        title: 'Earn',
        icon: <Icon name={ICON.ROCKET} size={FONT_SIZE.HUGE} />,
        inactiveIcon: <Icon name={ICON.ROCKET} size={FONT_SIZE.HUGE} color={COLOR.DARK_GRAY} />
      },
      [NavigationButtonType.REWARD]: {
        title: 'Rewards',
        icon: <Icon name={ICON.REWARDS_GIFT_CARDS} size={FONT_SIZE.HUGE} />,
        inactiveIcon: <Icon name={ICON.REWARDS_GIFT_CARDS} size={FONT_SIZE.HUGE} color={COLOR.DARK_GRAY} />
      },
      [NavigationButtonType.WALLET]: {
        title: 'Wallet',
        icon: <Icon name={ICON.CARD} size={FONT_SIZE.HUGE} />,
        inactiveIcon: <Icon name={ICON.CARD} size={FONT_SIZE.HUGE} color={COLOR.DARK_GRAY} />
      }
    }),
    []
  );

  const contentDataSet = contentMap[type];

  if (!contentDataSet) return null;

  return (
    <TouchableHighlight {...props} {...getTestIdProps(`${type}`)} onPress={onPress} style={styles.container} underlayColor={null}>
      <AnimatedView transition="opacity" style={[styles.buttonContainer, active && styles.active]} duration={200}>
        <View>{active ? contentMap[type].icon : contentMap[type].inactiveIcon}</View>
        <Text style={[styles.title, active && styles.textActive]}>{contentMap[type].title}</Text>
      </AnimatedView>
    </TouchableHighlight>
  );
};

export default memo(NavigationButton);
