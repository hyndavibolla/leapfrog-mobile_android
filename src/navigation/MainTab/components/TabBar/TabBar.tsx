import React, { memo, useMemo } from 'react';
import { View } from 'react-native';
import { useNavigation } from '@react-navigation/native';

import { NavigationButton } from '_components/NavigationButton';
import { NavigationButtonType } from '_components/NavigationButton/NavigationButton';
import { ROUTES } from '_constants';

import { styles } from './styles';

export type TabRoute = keyof typeof ROUTES.MAIN_TAB;

export interface Props {
  activeRoute: TabRoute;
}

export const TabBar = ({ activeRoute }: Props) => {
  const { navigate } = useNavigation();

  const tabList = useMemo(
    () =>
      [
        [NavigationButtonType.EARN, ROUTES.MAIN_TAB.EARN],
        [NavigationButtonType.STREAK, ROUTES.MAIN_TAB.STREAK],
        [NavigationButtonType.REWARD, ROUTES.MAIN_TAB.REWARDS],
        [NavigationButtonType.WALLET, ROUTES.MAIN_TAB.WALLET]
      ].filter(Boolean) as [NavigationButtonType, TabRoute][],
    []
  );

  return (
    <View style={styles.container}>
      {tabList.map(([type, route]) => (
        <NavigationButton key={type} type={type} active={route === activeRoute} onPress={() => navigate(route)} />
      ))}
    </View>
  );
};

export default memo(TabBar);
