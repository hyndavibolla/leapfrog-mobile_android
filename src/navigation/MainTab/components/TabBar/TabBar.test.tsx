import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import { NavigationButtonType } from '_components/NavigationButton/NavigationButton';
import { ROUTES } from '_constants';

import TabBar, { Props, TabRoute } from './TabBar';

const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
  __esModule: true,
  useNavigation: () => ({ navigate: mockNavigate })
}));

describe('TabBar', () => {
  let props: Props;
  beforeEach(() => {
    props = {
      activeRoute: ROUTES.MAIN as TabRoute
    };
  });
  it('should render', () => {
    const { toJSON } = render(<TabBar {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should navigate', async () => {
    const { findByTestId } = render(<TabBar {...props} />);
    const button = await findByTestId(`navigation-button-${NavigationButtonType.REWARD}`);
    fireEvent.press(button);
    expect(mockNavigate).toBeCalledWith(ROUTES.MAIN_TAB.REWARDS);
  });
});
