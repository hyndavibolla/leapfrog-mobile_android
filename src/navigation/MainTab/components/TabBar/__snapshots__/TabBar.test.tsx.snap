// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBar should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flexDirection": "row",
      "justifyContent": "space-around",
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 0,
    }
  }
>
  <View
    accessibilityLabel="navigation-button-earn"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
      }
    }
    testID="navigation-button-earn"
  >
    <div
      duration={200}
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 75,
            "justifyContent": "center",
            "width": 75,
          },
          false,
        ]
      }
      transition="opacity"
    >
      <View>
        <Text
          accessibilityLabel="icon-text"
          accessible={true}
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontSize": 26,
              },
              undefined,
              Object {
                "fontFamily": "sywmax",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="icon-text"
        >
          
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Medium",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Array [
              Object {
                "color": "#909BA0",
                "fontSize": 12,
                "marginTop": 8,
                "opacity": 0.5,
              },
              false,
            ],
          ]
        }
      >
        Earn
      </Text>
    </div>
  </View>
  <View
    accessibilityLabel="navigation-button-streak"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
      }
    }
    testID="navigation-button-streak"
  >
    <div
      duration={200}
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 75,
            "justifyContent": "center",
            "width": 75,
          },
          false,
        ]
      }
      transition="opacity"
    >
      <View>
        <Text
          accessibilityLabel="icon-text"
          accessible={true}
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontSize": 26,
              },
              undefined,
              Object {
                "fontFamily": "sywmax",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="icon-text"
        >
          
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Medium",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Array [
              Object {
                "color": "#909BA0",
                "fontSize": 12,
                "marginTop": 8,
                "opacity": 0.5,
              },
              false,
            ],
          ]
        }
      >
        Missions
      </Text>
    </div>
  </View>
  <View
    accessibilityLabel="navigation-button-reward"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
      }
    }
    testID="navigation-button-reward"
  >
    <div
      duration={200}
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 75,
            "justifyContent": "center",
            "width": 75,
          },
          false,
        ]
      }
      transition="opacity"
    >
      <View>
        <Text
          accessibilityLabel="icon-text"
          accessible={true}
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontSize": 26,
              },
              undefined,
              Object {
                "fontFamily": "sywmax",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="icon-text"
        >
          
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Medium",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Array [
              Object {
                "color": "#909BA0",
                "fontSize": 12,
                "marginTop": 8,
                "opacity": 0.5,
              },
              false,
            ],
          ]
        }
      >
        Rewards
      </Text>
    </div>
  </View>
  <View
    accessibilityLabel="navigation-button-wallet"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
      }
    }
    testID="navigation-button-wallet"
  >
    <div
      duration={200}
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 75,
            "justifyContent": "center",
            "width": 75,
          },
          false,
        ]
      }
      transition="opacity"
    >
      <View>
        <Text
          accessibilityLabel="icon-text"
          accessible={true}
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontSize": 26,
              },
              undefined,
              Object {
                "fontFamily": "sywmax",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="icon-text"
        >
          
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Medium",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Array [
              Object {
                "color": "#909BA0",
                "fontSize": 12,
                "marginTop": 8,
                "opacity": 0.5,
              },
              false,
            ],
          ]
        }
      >
        Wallet
      </Text>
    </div>
  </View>
</View>
`;
